### PR TITLE
feat(documentation-v7): migrate Select component stories

### PR DESCRIPTION
**Pull request by b1aserlu**

### CODE REVIEW

File: packages/documentation-v7/src/stories/components/select/select.docs.mdx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+<Canvas of={SelectStories.Default} />
+<div className="hide-col-default">
+  <Controls of={SelectStories.Default} />
+</div>
+
+## SCSS Imports
+
+To import all styles from the design-system-styles package:
+
+```scss dark
```
Comment (oliverschuerch) [inline packages/documentation-v7/src/stories/components/select/select.docs.mdx:24]:
```suggestion
```scss
```

File: packages/documentation-v7/src/stories/components/select/select.docs.mdx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+@use '@swisspost/design-system-styles/index.scss';
+```
+
+To import only the styles required for this component:
+
+```scss dark
```
Comment (oliverschuerch) [inline packages/documentation-v7/src/stories/components/select/select.docs.mdx:30]:
```suggestion
```scss
```

File: packages/documentation-v7/src/stories/components/select/select.stories.ts [hunk 3]
```diff
@@ -0,0 +1,328 @@
+import type { Args, Meta, StoryContext, StoryObj } from '@storybook/web-components';
+import { html, unsafeStatic } from 'lit/static-html.js';
+import { BADGE } from '../../../../.storybook/constants';
+import { LitElement } from 'lit';
+import { useArgs } from '@storybook/preview-api';
```
Comment (oliverschuerch) [inline packages/documentation-v7/src/stories/components/select/select.stories.ts:5]: Import node modules first (group them if they come from the same module) and custom imports second.

```suggestion
import { LitElement } from 'lit';
import { html, unsafeStatic } from 'lit/static-html.js';
import { useArgs } from '@storybook/preview-api';
import { BADGE } from '../../../../.storybook/constants';
```

File: packages/documentation-v7/src/stories/components/select/select.stories.ts [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+        @change=${(e: Event) => {
+          updateArgs({ value: (e.target as HTMLSelectElement).value });
+
+          if (document.activeElement === e.target) {
+            setTimeout(() => {
+              const element: HTMLSelectElement | null = document.querySelector(`#${id}`);
+              if (element) element.focus();
+            }, 25);
+          }
+        }}
```
Comment (oliverschuerch) [inline packages/documentation-v7/src/stories/components/select/select.stories.ts:241]: To reduce complexity, you can extract this function:
- Define the function somewhere beside the `control` definition, like so: `function onChange(e: Event) { ... }`
- Bind the function like so `@change="${onChange}"`.

**Review by oliverschuerch — CHANGES_REQUESTED** [review]

File: packages/documentation-v7/src/stories/components/select/select.stories.ts [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+    ];
+    function onChange(e: Event) {
+      updateArgs({ value: (e.target as HTMLSelectElement).value });
+
+      if (document.activeElement === e.target) {
+        setTimeout(() => {
+          const element: HTMLSelectElement | null = document.querySelector(`#${id}`);
+          if (element) element.focus();
+        }, 25);
+      }
```
Comment (oliverschuerch) [inline packages/documentation-v7/src/stories/components/select/select.stories.ts:229]: These lines are setting the focus programmatically to the select field. It's probably not needed anymore in Storybook version 7.
Remove them and check if the focus stays on the select after changing its value in the story. If this is the case, we can get rid of these lines and inline the onChange function again into the control as a one-liner:

```ts
@change="${(e: Event) => updateArgs({ value: (e.target as HTMLSelectElement).value })}"
```

**Review by oliverschuerch — CHANGES_REQUESTED** [review]

File: packages/documentation-v7/src/stories/components/select/select.stories.ts [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+    }
+    const control = html`
+      <select
+        id=${id}
+        class=${classes}
+        defaultValue=${args.value}
+        ?multiple=${args.multiple}
+        size=${args.multipleSize}
```
[the remark this answers is not in the document]
Comment (b1aserlu) [reply packages/documentation-v7/src/stories/components/select/select.stories.ts:237]: I found a way to do this here: [https://lit.dev/docs/templates/expressions/#removing-child]

`size=${args.mulipleSize ?? nothing }`

File: packages/documentation-v7/src/stories/components/select/select.stories.ts (hunk 5 above)
Comment (b1aserlu) [reply packages/documentation-v7/src/stories/components/select/select.stories.ts:229]: You are correct it works without this part of the function.

**Review by oliverschuerch — APPROVED** [review]

**Pull request merged**